### PR TITLE
Make more precise about dependencies of Sys.WORD_SIZE, default integer and larger integers.

### PR DESCRIPTION
**Pull request by nagexiucai**
come from #28868 which contains more than one logical/conceptual change.

### CODE REVIEW

**Review by fredrikekre — APPROVED** [review]

File: doc/src/manual/integers-and-floating-point-numbers.md [hunk 1]
```diff
@@ -56,28 +56,27 @@ julia> 1234
 1234
 ```
 
-The default type for an integer literal depends on whether the target system has a 32-bit architecture
-or a 64-bit architecture:
+The default type for an integer literal depends on version of julia, 32-bit or 64-bit:
```
Comment (StefanKarpinski) [inline doc/src/manual/integers-and-floating-point-numbers.md]: “depends on **the** version”

Comment (nagexiucai) [reply doc/src/manual/integers-and-floating-point-numbers.md]: thanks, oh my poor english.

**Review by StefanKarpinski — APPROVED** [review]

**Comment by JeffBezanson** [comment]
I suppose this is fine, but julia's word size is indeed tied to the architecture it was built on.

**Comment by StefanKarpinski** [comment]
This would still be accurate, however, if we allowed building a 32-bit Julia on a 64-bit system and vice versa, so this does seem slightly better to me.

**Comment by JeffBezanson** [comment]
That's true, but allowing that is not just a matter of flipping a switch. It seems like we should now explicitly mention that currently (and for the foreseeable future) julia's word size is equal to the system word size.

**Pull request closed without merging**